app/delivery/send_to_providers.py

```diff
@@ -211,12 +212,16 @@ def provider_to_use(notification_type, notification_id, international=False, sen
         )
         raise Exception("No active {} providers".format(notification_type))
 
-    if sender is not None and notification_type == SMS_TYPE and sender[0] == "+":
-        return clients.get_client_by_name_and_type("pinpoint", notification_type)
+    if _sms_send_on_pinpoint(notification_type, sender):
+        return clients.get_client_by_name_and_type(PINPOINT_PROVIDER, notification_type)
 
     return clients.get_client_by_name_and_type(active_providers_in_order[0].identifier, notification_type)
 
 
+def _sms_send_on_pinpoint(notification_type, sender):
```

Comment:
I don't really like this name, wondering if we could rename it to something else. In fact, we're not forced to send these using Pinpoint now, the logic is really "is it going to be sent with a dedicated phone number"

Reply:
What if we rename it to something like "send_dedicated_number_sms"?
And PINPOINT_PROVIDER could be "DEDICATED_NUMBER_PROVIDER"? Longer but more clear?